### PR TITLE
[wallet-refactor-5] Dumbify components, and move web3-react dependant to its directory

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

This PR is a big one. Sorry for that. 

## Main Refactor
Because an image speaks a thousand words, this PR tries to continue the refactor and do this:
image

Also, the connectors, have some parts that depend on web3-react, and some part that doesn't (in principle, we can use other library and provide other implementation)

* So colors, how they look, the names of the wallets, the URL they open, etc. ----> will not depend on 
* Deciding if is active, or activate the wallet ----> will be web3-react specific

So those parts are clearly separated
image

## Connectors / Connections
Additionally, now each connection option will have in the same file everything related to that option. 
* connection
* component with the connect button

Also, its easier to find all the options next to each other:
image

This way, adding a new wallet should be mostly self contained (mostly, there's a few parts on the code that still depends on importing connections and doing some logics with it, this will be addressed in other PR). 

image

## Types Web3ReactConnnection

The type Connnection was renamed to Web3ReactConnnection to signal its dependency with Web3React. 
Ideally, the app shouldn't depend in this type (although for now it does)
The new name will allow to quickly spot when we are using this out of place. 

In the same way `geConnection` has been replaced by `getWeb3ReactConnection`. This was used a lot in the app. And shouldn't.

image

## Move hooks
Lastly, the PR moves, and clean up some hooks related to connections which has dependencies with the wallets

image

# To Test
Everything should work as before.

# Side note
What a nice feeling to be able to start to kill some uniswap code. Let's try to do little by little:
- Move original and custom code into `cow-react` into the right module
- Unify Mods/index/original into simpler files
- Separate Pure/Containers
- Delete parts that are not used

### CODE REVIEW

**Review by shoom3301 — APPROVED** [review]

File: src/cow-react/modules/wallet/api/pure/StatusIcon/index.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { ConnectionType } from 'connection'
+import { ConnectionType } from '@cow/modules/wallet'
 import styled from 'styled-components/macro'
 
-import CoinbaseWalletIcon from '../../assets/coinbase.svg'
-import FortmaticIcon from '../../assets/formatic.png'
-import WalletConnectIcon from '../../assets/walletConnectIcon.svg'
-import { Identicon } from '../Identicon'
+import CoinbaseWalletIcon from '../../../api/assets/coinbase.svg'
```
Comment (nenadV91) [inline src/cow-react/modules/wallet/api/pure/StatusIcon/index.tsx:4]: Should we use absolute paths here?

**Review by alfetopito — APPROVED** [review]
I second Nenad's comment regarding relative paths

File: src/cow-react/modules/wallet/api/pure/StatusIcon/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/cow-react/modules/wallet/api/pure/StatusIcon/index.tsx:4]: in principle they are in the same module, so i guess relative is also fine. We could do it, i don't have a strong preference

File: src/cow-react/modules/wallet/web3-react/hooks/useEagerlyConnect.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import { Connector } from '@web3-react/types'
-import { gnosisSafeConnection, networkConnection } from 'connection'
-import { getConnection } from '@cow/modules/wallet/api/utils'
+import { networkConnection } from '@cow/modules/wallet/web3-react/connection/ network'
```
Comment (nenadV91) [inline src/cow-react/modules/wallet/web3-react/hooks/useEagerlyConnect.ts:2]: There is an empty space ` network`, I assume this is a typo?

**Comment by nenadV91** [comment]
Active wallet text doesn't have the correct styles. 
Probably an issue happened somewhere during the refactors.

Screenshot 2023-03-02 at 09 55 08

**Comment by elena-zh** [comment]
> Active wallet text doesn't have the correct styles. Probably an issue happened somewhere during the refactors.
> 
> Screenshot 2023-03-02 at 09 55 08

Reported in #2070

**Review by nenadV91 — APPROVED** [review]
Apart from those minor comments, LGTM

File: src/cow-react/modules/wallet/web3-react/hooks/useEagerlyConnect.ts (hunk 2 above)
Comment (anxolin) [reply src/cow-react/modules/wallet/web3-react/hooks/useEagerlyConnect.ts]: Weird, somehow i manage to pre-pend the name of the file with a space! the import was working like that. amazing!
It's fixed here f693626389a33ae47c0c628444e9be79e8e70c32

File: src/cow-react/modules/wallet/api/pure/StatusIcon/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/cow-react/modules/wallet/api/pure/StatusIcon/index.tsx]: Addressed

**Pull request merged**